src/services/symbolDisplay.ts

```diff
@@ -89,7 +89,15 @@ namespace ts.SymbolDisplay {
                 case SyntaxKind.JsxOpeningElement:
                 case SyntaxKind.JsxElement:
                 case SyntaxKind.JsxSelfClosingElement:
-                    return location.kind === SyntaxKind.Identifier ? ScriptElementKind.memberVariableElement : ScriptElementKind.jsxAttribute;
+                    switch (location.kind) {
+                        case SyntaxKind.Identifier:
+                            return ScriptElementKind.memberVariableElement;
+                        case SyntaxKind.LessThanToken:
+                        case SyntaxKind.SyntaxList:
+                            return ScriptElementKind.jsxTagName;

```

Comment:
It would be good to test these for `<Foo.Bar className="" />` and `<Foo<Bar> className="" />`. Probably already ok due to different parents, but good to check for good measure.

Reply:
Good point. Will write some tests.

Reply:
The latter case doesn't work before or after at all, actually:

![image](https://user-images.githubusercontent.com/5341706/148136874-137fe1db-a220-4389-baac-108a3c34d50b.png)

Whereas, omit the type param and you get the completion:

![image](https://user-images.githubusercontent.com/5341706/148136911-dedf3654-6f39-43be-9df9-c9bc00c37e13.png)

Not a regression, but I'll file a new issue once I'm finished with a test case.

Reply:
Your first example doesn't work with this PR, it turns out; the joy of the `location` being the same for both `Foo.|` and `Foo.Foo |`.

The screenshot above is misleading in that I took it while the original code was there (when I was verifying that my test failed when run on the old code, but before I actually wrote the second test). I'll keep looking.

Reply:
Filed the above as #47416, for reference.